README.md

```diff
@@ -4,130 +4,183 @@
+    S4TF's advantages include a seamless integration with a modern general-purpose
+    language, allowing for more dynamic and sophisticated models. Fast
+    abstractions can be developed "in user-space" (as opposed to in C/C++
+    aka "framework-space"), resulting in modular APIs that can be easily
+    customized.
+
+ 2. **ML learners** who are just getting started with machine learning. Thanks
+    to Swift's support for quality tooling (e.g. context-aware autocomplete),
+    Swift for TensorFlow can be one of the most productive ways to get started
+    learning the fundamentals of machine learning.
+
+## Getting started
+
+ - **Google Colaboratory**: The fastest way to get started is to try out S4TF right in your
+   browser. Just open up our [getting started
+   notebook](https://colab.research.google.com/github/tensorflow/swift-tutorials/blob/master/iris/swift_tensorflow_tutorial.ipynb) (or start from a
+   [blank notebook](https://colab.research.google.com/github/tensorflow/swift/blob/master/notebooks/blank_swift.ipynb))!
+   Read more in our [usage guide](Usage.md).
+
+ - **Install locally**: you can [download a pre-built Swift for TensorFlow
+   package](Installation.md). After installation, you can follow these
+   [step-by-step instructions](Usage.md) to build and execute a Swift script
+   on your computer.
+
+ - **Compile from source**: If you'd like to customize S4TF or even contribute
+   back, follow our [instructions on building the S4TF compiler from
+   source](https://github.com/apple/swift/tree/tensorflow).
+
+Please do join the
+[swift@tensorflow.org mailing list](https://groups.google.com/a/tensorflow.org/d/forum/swift)
+to hear the latest announcements, get help, and share your thoughts!
+
+## Why Swift for TensorFlow?
 
 Swift for TensorFlow is a new way to develop machine learning models. It
 gives you the power of
 [TensorFlow](https://www.tensorflow.org) directly integrated into the
-[Swift programming language](https://swift.org/about).
-With Swift, you can write the following imperative code, and Swift 
-automatically turns it into **a single TensorFlow Graph** and runs it 
-with the full performance of TensorFlow Sessions on CPU, GPU and 
-[TPU](https://cloud.google.com/tpu).
+[Swift programming language](https://swift.org/about). We believe that
+machine learning paradigms are so important that they deserve
+**first-class language and compiler support**. 
 
-```swift
-import TensorFlow
+A fundamental primitive in machine learning is gradient-based optimization:
+computing function derivatives to optimize parameters. With Swift for
+TensorFlow, you can easily differentiate custom functions using differential
+operators like `gradient(of:)`, or differentiate with respect to an entire
+model by calling standard library method `model.gradient { ... }`.
 
-var x = Tensor<Float>([[1, 2], [3, 4]])
+```swift
+// Custom differentiable type.
+struct Model : Differentiable {
```

Comment:
In public code, we use 4-space indentation and the "no space before any colon" style.

```suggestion
struct Model: Differentiable {
```

Reply:
Ack; fixed.